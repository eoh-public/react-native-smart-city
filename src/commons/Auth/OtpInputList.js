import React, { useState, useEffect } from 'react';
import { View, TextInput, StyleSheet } from 'react-native';

import { Theme, Colors } from '../../configs';

const OtpInputList = ({ onfinishInputCode }) => {
  let firstTextInput;
  let secondTextInput;
  let thirdTextInput;
  let fourthTextInput;
  let fifthTextInput;
  let sixthTextInput;
  const [verificationCodeArray, setVerificationCodeArray] = useState([
    '',
    '',
    '',
    '',
    '',
    '',
  ]);

  const inputFocusHandler = (index) => {
    verificationCodeArray.splice(index, 1, ''); // replace 1 element with empty string at [index] of array
    setVerificationCodeArray((prev) => [...verificationCodeArray]);
    switch (index) {
      case 0:
        firstTextInput.clear();
        firstTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
      case 1:
        secondTextInput.clear();
        secondTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
      case 2:
        thirdTextInput.clear();
        thirdTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
      case 3:
        fourthTextInput.clear();
        fourthTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
      case 4:
        fifthTextInput.clear();
        fifthTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
      case 5:
        sixthTextInput.clear();
        sixthTextInput.setNativeProps({
          ...Theme.transparentBackgroundOrangeBorder,
        });
        break;
    }
  };

  const inputChangeTextHandler = (index, text) => {
    if (text.length > 0) {
      verificationCodeArray.splice(index, 1, text); // replace 1 element at [index] of array
      setVerificationCodeArray((prev) => [...verificationCodeArray]);
    }
    switch (index) {
      case 0:
        if (text.length > 0) {
          firstTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        secondTextInput.focus();
        break;
      case 1:
        if (text.length > 0) {
          secondTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        thirdTextInput.focus();
        break;
      case 2:
        if (text.length > 0) {
          thirdTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        fourthTextInput.focus();
        break;
      case 3:
        if (text.length > 0) {
          fourthTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        fifthTextInput.focus();
        break;
      case 4:
        if (text.length > 0) {
          fifthTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        sixthTextInput.focus();
        break;
      case 5:
        if (text.length > 0) {
          sixthTextInput.setNativeProps({
            ...Theme.transparentBackgroundGrayBorder,
          });
        }
        break;
    }
  };

  useEffect(() => {
    let finishInputCode = true;
    let fullVerificationCode = '';
    verificationCodeArray.map((code) => {
      if (code === '') {
        finishInputCode = false;
      } else {
        fullVerificationCode += code;
      }
    });
    if (finishInputCode && typeof onfinishInputCode === 'function') {
      onfinishInputCode(fullVerificationCode);
    }
  }, [verificationCodeArray, onfinishInputCode]);

  return (
    <View style={styles.otpInputContainer}>
      <TextInput
        style={{
          ...styles.otpInput,
        }}
        name="firstInput"
        maxLength={1}
        onFocus={() => inputFocusHandler(0)}
        onChangeText={(text) => inputChangeTextHandler(0, text)}
        ref={(input) => (firstTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />
      <TextInput
        style={{
          ...styles.otpInput,
        }}
        maxLength={1}
        onFocus={() => inputFocusHandler(1)}
        onChangeText={(text) => inputChangeTextHandler(1, text)}
        ref={(input) => (secondTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />
      <TextInput
        style={{
          ...styles.otpInput,
        }}
        maxLength={1}
        onFocus={() => inputFocusHandler(2)}
        onChangeText={(text) => inputChangeTextHandler(2, text)}
        ref={(input) => (thirdTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />

      <TextInput
        style={{
          ...styles.otpInput,
        }}
        maxLength={1}
        onFocus={() => inputFocusHandler(3)}
        onChangeText={(text) => inputChangeTextHandler(3, text)}
        ref={(input) => (fourthTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />
      <TextInput
        style={{
          ...styles.otpInput,
        }}
        maxLength={1}
        onFocus={() => inputFocusHandler(4)}
        onChangeText={(text) => inputChangeTextHandler(4, text)}
        ref={(input) => (fifthTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />
      <TextInput
        style={{
          ...styles.otpInput,
        }}
        maxLength={1}
        onFocus={() => inputFocusHandler(5)}
        onChangeText={(text) => inputChangeTextHandler(5, text)}
        ref={(input) => (sixthTextInput = input)}
        keyboardType={'number-pad'}
        returnKeyType={'done'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  otpInputContainer: {
    flexDirection: 'row',
    marginTop: 10,
  },
  otpInput: {
    width: 36,
    height: 36,
    textAlign: 'center',
    marginStart: 5,
    marginEnd: 5,
    borderRadius: 5,
    ...Theme.transparentBackgroundOrangeBorder,
    color: Colors.Gray9,
  },
});

export default OtpInputList;
