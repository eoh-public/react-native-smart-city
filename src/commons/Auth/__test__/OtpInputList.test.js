import React from 'react';
import { TextInput } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import OtpInputList from '../OtpInputList';

describe('Test OtpInputList', () => {
  const mockFunc = jest.fn();
  const list_code = [0, 1, 2, 3, 4, 5];
  let tree;

  for (const code of list_code) {
    test(`create OtpInputList ${code} onfinishInputCode is func`, () => {
      act(() => {
        tree = renderer.create(<OtpInputList onfinishInputCode={mockFunc} />);
      });

      const instance = tree.root;
      const textInputs = instance.findAllByType(TextInput);
      textInputs.forEach((el, index) => {
        act(() => {
          el.props.onFocus(code); // TODO check onFocus(code)
          el.props.onChangeText('1');
        });
      });
      expect(mockFunc).toHaveBeenCalledWith('111111');
    });
  }

  test('create OtpInputList not parameter', () => {
    act(() => {
      tree = renderer.create(<OtpInputList />);
    });
    const instance = tree.root;
    const textInputs = instance.findAllByType(TextInput);
    textInputs.forEach((el, index) => {
      act(() => {
        el.props.onFocus();
        el.props.onChangeText('');
      });
    });
    expect(mockFunc).toHaveBeenCalledWith('111111');
  });
});
