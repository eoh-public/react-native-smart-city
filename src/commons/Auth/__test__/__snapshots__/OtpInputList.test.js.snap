// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test OtpInputList create OtpInputList 0 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList 1 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList 2 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList 3 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList 4 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList 5 onfinishInputCode is func 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Test OtpInputList create OtpInputList not parameter 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    name="firstInput"
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    maxLength={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#262626",
        "height": 36,
        "marginEnd": 5,
        "marginStart": 5,
        "textAlign": "center",
        "width": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
